File: src/main/java/ru/vood/generator/generate/runner/TemplateEngine.java
package ru.vood.generator.generate.runner;

public enum TemplateEngine {

    FREE_MARKER(FreeMarkerEngine.class),
    FREE_MARKER_DATABASE(FreeMarkerDatabaseEngine.class),
    VELOCITY(VelocityEngine.class),
    ERROR(ErrorEngine.class);

    private final Class aClass;

    <T extends RunnerEngine> TemplateEngine(Class<T> runnerEngine) {
        aClass = runnerEngine;
    }

    public Class getaClass() {
        return aClass;
    }
}

File: src/main/java/ru/vood/generator/generate/runner/TemplateEngine.kt
package ru.vood.generator.generate.runner

import ru.vood.freemarker.FtlProcessor


enum class TemplateEngine(val runner: RunnerEngine) {
    FREE_MARKER(FreeMarkerEngine(FtlProcessor())),
    FREE_MARKER_DATABASE(FreeMarkerDatabaseEngine()),
    VELOCITY(VelocityEngine()),
    ERROR(ErrorEngine())
}
